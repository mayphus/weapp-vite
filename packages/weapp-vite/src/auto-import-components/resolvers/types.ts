export interface ResolvedValue { name: string, from: string }

export type Resolver = (componentName: string) => ResolvedValue | void

export type CreateResolver<T = any> = (options?: T) => Resolver
