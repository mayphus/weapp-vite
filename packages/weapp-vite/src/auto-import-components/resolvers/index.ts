export * from './tdesign'
export * from './types'
export * from './vant'
