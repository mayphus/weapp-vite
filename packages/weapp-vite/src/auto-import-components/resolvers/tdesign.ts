import type { CreateResolver } from './types'

// https://tdesign.tencent.com/miniprogram/components/button
export const TDesignResolver: CreateResolver = () => {
  const prefix = 't-'
  const components: string[] = [
    // 基础
    'button',
    'divider',
    'fab',
    'icon',
    'row',
    'col',
    'link',
    // 导航
    'back-top',
    'drawer',
    'indexes',
    'indexes-anchor',
    'navbar',
    'side-bar',
    'side-bar-item',
    'steps',
    'step-item',
    'tab-bar',
    'tab-bar-item',
    'tabs',
    'tab-panel',
    // 输入
    'calendar',
    'cascader',
    'checkbox',
    'checkbox-group',
    'color-picker',
    'date-time-picker',
    'input',
    'picker',
    'picker-item',
    'radio',
    'radio-group',
    'rate',
    'search',
    'slider',
    'stepper',
    'switch',
    'textarea',
    'tree-select',
    'upload',
    // 数据展示
    'avatar',
    'avatar-group',
    'badge',
    'cell',
    'cell-group',
    'collapse',
    'collapse-panel',
    'count-down',
    'empty',
    'footer',
    'grid',
    'grid-item',
    'image',
    'image-viewer',
    'progress',
    'result',
    'skeleton',
    'sticky',
    'swiper',
    'swiper-nav',
    'tag',
    'check-tag',
    // 反馈
    'action-sheet',
    'dialog',
    'dropdown-menu',
    'dropdown-item',
    'guide',
    'loading',
    'message',
    'notice-bar',
    'overlay',
    'popup',
    'pull-down-refresh',
    'swipe-cell',
  ]
  const map = components.reduce<Record<string, string>>((acc, cur) => {
    acc[`${prefix}${cur}`] = `tdesign-miniprogram/${cur}/${cur}`
    return acc
  }, {})
  return (componentName) => {
    const from = map[componentName]
    if (from) {
      return {
        name: componentName,
        from,
      }
    }
  }
}
