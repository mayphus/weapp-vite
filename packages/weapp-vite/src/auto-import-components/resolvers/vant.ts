import type { CreateResolver, Options } from './types'
import { defu } from '@weapp-core/shared'
import components from './json/vant.json'

const defaultPrefix = 'van-'
// https://vant-ui.github.io/vant-weapp/#/home
export const VantResolver: CreateResolver = (opts) => {
  const { prefix, resolve } = defu<Required<Options>, Options[]>(opts, {
    prefix: defaultPrefix,
    resolve({ name, prefix }) {
      return {
        key: `${prefix}${name}`,
        // 最后 + /index 似乎有问题
        value: `@vant/weapp/${name}`,
      }
    },
  })

  const map = components.reduce<Record<string, string>>((acc, cur) => {
    const { key, value } = resolve({
      name: cur,
      prefix,
    })

    acc[key] = value
    return acc
  }, {})
  return (componentName) => {
    const from = map[componentName]
    if (from) {
      return {
        name: componentName,
        from,
      }
    }
  }
}
