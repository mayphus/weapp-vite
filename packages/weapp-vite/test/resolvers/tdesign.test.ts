import { TDesignResolver } from '@/auto-import-components/resolvers/tdesign'

describe('tdesign', () => {
  it('tdesign default', () => {
    expect(TDesignResolver()).toMatchSnapshot()
  })
})
