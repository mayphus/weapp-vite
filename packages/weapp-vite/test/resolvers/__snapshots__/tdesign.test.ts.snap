// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tdesign > tdesign default 1`] = `[Function]`;
